scripts/uniqify-module-names.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+import json
+import argparse
+import shutil
+import os
+import sh
+
+
+
+parser = argparse.ArgumentParser(description="")
+parser.add_argument("--top-filelist", type=str, required=True, help="Abs path to <top>.<model>.top.f")
+parser.add_argument("--mod-filelist", type=str, required=True, help="Abs path to <top>.<model>.model.f")
+parser.add_argument("--gen-collateral-path", dest="gcpath", type=str, required=True, help="Abs path to the gen-collateral directory")
+parser.add_argument("--model-hier-json", type=str, required=True, help="Path to hierarchy JSON emitted by firtool. Must include DUT as a module.")
+parser.add_argument('--dut', type=str, required=True, help='Name of the DUT module.')
+args = parser.parse_args()
+
+
+def get_filelist(filelist):
+  f = open(filelist, "r")
+  lines = f.readlines()
+  f.close()
+
+  fnames = []
+  for line in lines:
+    try:
+      fname = line.split("/")[-1].replace("\n", "")
+      fnames.append(fname)
+    except:
+      print(f"Something is wrong about this line {line}")
+
+  return fnames
+
+def update_filelist(cur_file, new_file):
+  sh.sed("-i", f"s/\b{cur_file}\b/{new_file}/", os.path.join(args.gcpath, args.mod_filelist))
```

Comment:
Is there any issues with this and paths with `/`? If so, maybe we should use a different delimiter (`;`?) for `sed`

Reply:
can you explain what you mean by issues with paths with `/`?

Reply:
The default delimiter of `sed` is `/`, so if you are replacing things that have a `/` within them the `sed` will break. For example, if you want to replace `/hi/this/is/abe` with `/hi/this/is/joonho` the following `sed` command would break: `sed -i s//hi/this/is/abe//hi/this/is/joonho/g ...`. So instead you could do something like `sed -i 's#/hi/this/is/abe#/hi/this/is/joonho#g' ...` (with `#` as the new delimiter)

Reply:
Oh, but the `cur_file` & `new_file` here are just files (not paths) so I assume this is okay?